src/message.js

```diff
@@ -36,494 +36,610 @@ import config from './config';
 import crypto from './crypto';
 import enums from './enums';
 import util from './util';
-import packet from './packet';
+import {
+  PacketList,
+  LiteralDataPacket,
+  CompressedDataPacket,
+  AEADEncryptedDataPacket,
+  SymEncryptedIntegrityProtectedDataPacket,
+  SymmetricallyEncryptedDataPacket,
+  PublicKeyEncryptedSessionKeyPacket,
+  SymEncryptedSessionKeyPacket,
+  OnePassSignaturePacket,
+  SignaturePacket
+} from './packet';
 import { Signature } from './signature';
 import { getPreferredHashAlgo, getPreferredAlgo, isAeadSupported, createSignaturePacket } from './key';
 
 
 /**
- * @class
- * @classdesc Class that represents an OpenPGP message.
+ * Class that represents an OpenPGP message.
  * Can be an encrypted message, signed message, compressed message or literal message
- * @param  {module:packet.List} packetlist The packets that form this message
  * See {@link https://tools.ietf.org/html/rfc4880#section-11.3}
  */
-
-export function Message(packetlist) {
-  if (!(this instanceof Message)) {
-    return new Message(packetlist);
+export class Message {
```

Comment:
In my opinion it would be improvement to split Message file to Message class and other functions. That way one could use `export default class` here.

Reply:
Yeah, that could make sense :+1: I guess to be consistent, we'd have to do the same for files like `src/signature.js` though, which are already very small. I'll look into it.